src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -446,24 +446,26 @@ private object GetVSSolutionProperty(int propId)
             return value;
         }
 
-        private void OnSolutionExistsAndFullyLoaded()
+        private async Task<int> OnSolutionExistsAndFullyLoadedAsync(CancellationToken token)
         {
-            ThreadHelper.ThrowIfNotOnUIThread();
+            await TaskScheduler.Default;
```

Comment:
why you need to explicitly go to bg thread? this is a private api and is being called on bg thread itself in all the cases...